x-pack/plugins/infra/public/containers/logs/log_entries/index.ts

```diff
@@ -191,46 +262,72 @@ const useFetchEntriesEffect = (
 
   const fetchNewerEntries = useCallback(
     throttle(() => runFetchMoreEntriesRequest(ShouldFetchMoreEntries.After), 500),
-    [props, state.entriesEnd]
+    [props, state.bottomCursor]
   );
 
   const streamEntriesEffectDependencies = [
-    props.isAutoReloading,
+    props.isStreaming,
     state.isLoadingMore,
     state.isReloading,
   ];
   const streamEntriesEffect = () => {
     (async () => {
-      if (props.isAutoReloading && !state.isLoadingMore && !state.isReloading) {
+      if (props.isStreaming && !state.isLoadingMore && !state.isReloading) {
         if (startedStreaming) {
-          await new Promise(res => setTimeout(res, 5000));
+          await new Promise(res => setTimeout(res, LIVE_STREAM_INTERVAL));
         } else {
-          const nowKey = {
-            tiebreaker: 0,
-            time: Date.now(),
-          };
-          props.jumpToTargetPosition(nowKey);
+          const endTimestamp = Date.now();
+          props.jumpToTargetPosition({ tiebreaker: 0, time: endTimestamp });
           setStartedStreaming(true);
           if (state.hasMoreAfterEnd) {
-            runFetchNewEntriesRequest({
-              timeKey: nowKey,
-            });
+            runFetchNewEntriesRequest({ endTimestamp });
             return;
           }
         }
         const newEntriesEnd = await runFetchMoreEntriesRequest(ShouldFetchMoreEntries.After);
         if (newEntriesEnd) {
           props.jumpToTargetPosition(newEntriesEnd);
         }
-      } else if (!props.isAutoReloading) {
+      } else if (!props.isStreaming) {
         setStartedStreaming(false);
       }
     })();
   };
 
+  const expandRangeEffect = () => {
+    const shouldExpand = {
+      before: false,
+      after: false,
+    };
+
+    if (!prevParams || !prevParams.startTimestamp || !prevParams.endTimestamp) {
+      return;
+    }
+
+    if (props.timestampsLastUpdate === prevParams.timestampsLastUpdate) {
+      return;
+    }
+
+    if (props.startTimestamp < prevParams.startTimestamp) {
+      shouldExpand.before = true;
+    }
+    if (props.endTimestamp > prevParams.endTimestamp) {
+      shouldExpand.after = true;
+    }
+
+    dispatch({ type: Action.ExpandRange, payload: shouldExpand });
+  };
+
+  const expandRangeEffectDependencies = [
```

Comment:
I get feedback similar to this a lot :) I need to change my mental model.

`prevProps` can only change if we get new `props`, so in my head adding `prevProps` as a dependency is redundant. Of course _I_ know this, because _I_ coded it, but it might not be obvious without more context. A fresh reader now needs to track that indeed `prevProps` only changes with new props.

I can add `prevProps?.startTimestamp` and `prevProps?.endTimestamp` as dependencies. `prevProps` can be null so we need to check for that, but I also don't want this to trigger if any prop different that the timestamps change.

Makes sense?